numba/core/bytecode.py

```diff
@@ -321,45 +363,199 @@ def get_used_globals(self):
 
 
 def _fix_LOAD_GLOBAL_arg(arg):
-    if PYVERSION >= (3, 11):
-        assert PYVERSION == (3, 11) # reminder to check newer versions
+    if PYVERSION in ((3, 11), (3, 12)):
         return arg >> 1
-    return arg
+    elif PYVERSION in ((3, 8), (3, 9), (3, 10)):
+        return arg
+    else:
+        raise NotImplementedError(PYVERSION)
 
 
 class ByteCodePy311(_ByteCode):
+
     def __init__(self, func_id):
         super().__init__(func_id)
-
-        def fixup_eh(ent):
-            from dis import _ExceptionTableEntry
-            # Patch up the exception table offset
-            # because we add a NOP in _patched_opargs
-            out = _ExceptionTableEntry(
-                start=ent.start + _FIXED_OFFSET, end=ent.end + _FIXED_OFFSET,
-                target=ent.target + _FIXED_OFFSET,
-                depth=ent.depth, lasti=ent.lasti,
-            )
-            return out
-
         entries = dis.Bytecode(func_id.code).exception_entries
-        self.exception_entries = tuple(map(fixup_eh, entries))
+        self.exception_entries = tuple(map(self.fixup_eh, entries))
+
+    @staticmethod
+    def fixup_eh(ent):
+        # Patch up the exception table offset
+        # because we add a NOP in _patched_opargs
+        out = dis._ExceptionTableEntry(
+            start=ent.start + _FIXED_OFFSET, end=ent.end + _FIXED_OFFSET,
+            target=ent.target + _FIXED_OFFSET,
+            depth=ent.depth, lasti=ent.lasti,
+        )
+        return out
 
     def find_exception_entry(self, offset):
         """
         Returns the exception entry for the given instruction offset
         """
         candidates = []
         for ent in self.exception_entries:
-            if ent.start <= offset <= ent.end:
+            if ent.start <= offset < ent.end:
                 candidates.append((ent.depth, ent))
         if candidates:
             ent = max(candidates)[1]
             return ent
 
 
+class ByteCodePy312(ByteCodePy311):
+
+    def __init__(self, func_id):
+        super().__init__(func_id)
+
+        # initialize lazy property
+        self._ordered_offsets = None
+
+        # Fixup offsets for all exception entries.
+        entries = [self.fixup_eh(e) for e in
+                   dis.Bytecode(func_id.code).exception_entries
+                   ]
+
+        # Remove exceptions, innermost ones first
+        # Can be done by using a stack
+        entries = self.remove_build_list_swap_pattern(entries)
+
+        # If this is a generator, we need to skip any exception table entries
+        # that point to the exception handler with the highest offset.
```

Comment:
Is it always the highest offset (it seems to be in a few things I've tested)? Would it be a good idea to make sure that the `max_exception_target` is a `CALL_INTRINSIC_1(INTRINSIC_STOPITERATION_ERROR)`?